GVRf/Framework/src/org/gearvrf/script/GVRScriptFile.java

```diff
@@ -0,0 +1,151 @@
+/* Copyright 2016 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.script;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import javax.script.Bindings;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+
+import org.gearvrf.GVRContext;
+
+/**
+ * This class represents a script file, which can be attached to an
+ * object to handle events delivered to it. <p>
+ *
+ * It can be used in two ways:
+ * <ul>
+ *   <li> Use {@link GVRScriptManager#loadScript(org.gearvrf.GVRAndroidResource, String)
+ *   to load using a {@code GVRAndroidResource} object.</li>
+ *   <li> Construct the {@code GVRScriptFile} object and set the script
+ *   text using {@link #setScriptText(String)}, or load it from a stream using
+ *   {@link #load(InputStream)}. </li>
+ * </ul>
+ */
+public abstract class GVRScriptFile {
+    private static final String TAG = GVRScriptFile.class.getSimpleName();
+    GVRContext mGvrContext;
+    protected String mLanguage;
+    protected String mScriptText;
+    protected ScriptEngine mLocalEngine; 
+
+    /**
+     * Constructor.
+     *
+     * @param gvrContext
+     *     The GVR Context.
+     * @param language
+     *     The language of the script file. Please use the constants
+     *     {@code LANG_*} defined in {@link GVRScriptManager}, such
+     *     as {@code LANG_LUA}, {@code LANG_JAVASCRIPT}, and so on.
+     */
+    public GVRScriptFile(GVRContext gvrContext, String language) {
+        mGvrContext = gvrContext;
+        mLanguage = language;
+        ScriptEngine engine = mGvrContext.getScriptManager().getEngine(mLanguage);
+        mLocalEngine = engine.getFactory().getScriptEngine();
+
+        // Add globals
+        mGvrContext.getScriptManager().addGlobalBindings(mLocalEngine);
+    }
+
+    /**
+     * Loads a script into the {@GVRScriptFile} object.
+     *
+     * @param inputStream
+     *     The input stream from which to load the script.
+     * @throws IOException
+     */
+    public void load(InputStream inputStream) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
```

Comment:
Good question. I checked into this a bit, and can confirm that "when the reader is closed, it will close the underlying InputStream". This is a side-effect I didn't want. The creator of the InputStream should close it instead?